mmdet/apis/inference.py

```diff
@@ -131,21 +142,20 @@ def inference_detector(
             data_ = dict(img_path=img, img_id=0)
         # build the data pipeline
         data_ = test_pipeline(data_)
-        data.append(data_)
 
-    for m in model.modules():
-        assert not isinstance(
-            m,
-            RoIPool), 'CPU inference with RoIPool is not supported currently.'
+        data_['inputs'] = [data_['inputs']]
```

Comment:
Can this be replaced with  collate_fn here?

Reply:
Yes, but is it necessary?